test/blackbox-tests/test-cases/watching/copy-rules.t

```diff
@@ -48,23 +48,22 @@ consequence of using a glob in this directory, which forces all *.txt rules.
   b
   c
 
-Now demonstrate a bug: Dune fails to notice that [d.txt] appears via promotion.
-
-# CR-someday amokhov: Fix this test.
+Dune notices that [d.txt] appears via promotion.
 
   $ mkdir subdir
   $ cat > subdir/dune <<EOF
   > (rule
   >   (target d.txt)
   >   (mode (promote (into ..)))
-  >   (action (write-file %{target} d)))
+  >   (action (write-file %{target} "d\n")))
   > EOF
   $ build summary subdir/d.txt
   Success
   $ cat _build/default/summary
   a
   b
   c
+  d
```

Comment:
This test appears to be non-deterministic: it works for me locally but on CI this last `d` is missing, hinting that Dune somehow was too late to react to this change. I'm going to add a `sleep` here to try fixing this.

Reply:
Oh, adding `sleep` helped.

Reply:
We might no longer need this sleep now that we don't re-order events

Reply:
Indeed, removing because #5154 should hopefully fix the race.